packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,231 @@
+import debounce from 'debounce-promise';
+import differenceWith from 'lodash.differencewith';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import { useIntl } from 'react-intl';
+import { DragDropContext } from 'react-beautiful-dnd';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import CollapsibleMotion from '@commercetools-uikit/collapsible-motion';
+import FieldLabel from '@commercetools-uikit/field-label';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import {
+  CloseIcon,
+  EyeCrossedIcon,
+  EyeIcon,
+  SearchIcon,
+} from '@commercetools-uikit/icons';
+import AccessibleButton from '@commercetools-uikit/accessible-button';
+import { filterDataAttributes } from '@commercetools-uikit/utils';
+import DroppablePanel from '../droppable-panel';
+import {
+  Container,
+  HeaderContainer,
+  DroppableContainer,
+} from '../data-table-settings-panel.styles';
+import messages from './messages';
+
+export const ColumnManager = (props) => {
+  const { formatMessage } = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+  const handleDragEnd = (result) => {
```

Comment:
This `handleDragEnd` and the below `handleDragStart` helpers could be extracted from the component definition so that their logic could be tested in the associated spec.